service/src/main/java/org/folio/rest/workflow/service/WorkflowEngineService.java

```diff
@@ -52,15 +55,41 @@ public Workflow deactivate(String workflowId, String tenant, String token) throw
     return sendRequest(workflow, WORKFLOW_ENGINE_DEACTIVATE_URL_TEMPLATE, tenant, token);
   }
 
+  public JsonNode start(String workflowId, String tenant, String token, JsonNode context) throws WorkflowEngineServiceException {
+    Workflow workflow = workflowRepo.getOne(workflowId);
+
+    JsonNode definition = fetchDefinition(workflow.getDeploymentId(), tenant, token);
+    String definitionId = definition.get("id").asText();
+
+    HttpEntity<JsonNode> contextHttpEntity = new HttpEntity<>(context, headers(tenant, token));
+
+    String url = String.format("%s/camunda/process-definition/%s/start", okapiUrl, definitionId);
+    ResponseEntity<JsonNode> response = exchange(url, HttpMethod.POST, contextHttpEntity, JsonNode.class);
+
+    return response.getBody();
+  }
+
+  private JsonNode fetchDefinition(String deploymentId, String tenant, String token) throws WorkflowEngineServiceException {
+    String definitionsUrl = String.format("%s/camunda/process-definition?deploymentId=%s", okapiUrl, deploymentId);
+    ResponseEntity<ArrayNode> response = this.restTemplate.getForEntity(definitionsUrl, ArrayNode.class);
+
+    if (response.getStatusCode() == HttpStatus.OK && !response.getBody().isEmpty()) {
+      logger.debug("Response body: {}", response.getBody());
+
+      try {
+        return response.getBody().get(0);
+      } catch (Exception e) {
+
```

Comment:
Should this have a `logger.debug()` call in the exception?

Reply:
Nah, the debug log is handled in the controller advise.